### ISSUE TITLE
simple_switch fails silently, possibly on testing struct equality

### ISSUE THREAD

**Issue by vikramnathan** [p4lang/behavioral-model#265]
For following p4 (which compiles into the attached json using p4c-bm2-ss):
[test_bug2.json.txt](https://github.com/p4lang/behavioral-model/files/711468/test_bug2.json.txt)
[test_bug2.p4.txt](https://github.com/p4lang/behavioral-model/files/711469/test_bug2.p4.txt)

simple_switch fails silently when running the following command:
```
$> sudo $BM/targets/simple_switch/simple_switch -i 0@veth2 -i 1@veth4 --log-file switch.log --log-level trace --log-flush --pcap test_bug2.json
Thrift port was not specified, will use 9090
Calling target program-options parser
$>
```
Observed: The switch fails silently before adding the interfaces. The log file is empty, regardless of the log level I use.
Expected: the switch starts and waits for input. Log file shows that interfaces were successfully added.

Note: when I change line 84 to test the equality of the struct *fields*, i.e. `bool same = (inKey.field1 == defaultKey.field1)`, the switch runs as expected.

**Comment by antoninbas** [comment]
I suspect the `p4c-bm2-ss` could be fixed to produce a correct JSON (i.e. the struct equality could be replaced with a sequence of field-by-field comparisons). However in this specific case I believe it would be beneficial to enhance the bmv2 expression engine to support struct (header) equality, so I will try to implement it.

**Comment by antoninbas** [comment]
fixed by #267
